projects/kit/components/slider/slider-readonly.directive.ts

```diff
@@ -1,41 +1,60 @@
-import {Directive, HostListener, Input} from '@angular/core';
-import {tuiDefaultProp} from '@taiga-ui/cdk';
+import {
+    ChangeDetectorRef,
+    Directive,
+    HostListener,
+    Inject,
+    Input,
+    Optional,
+    Self,
+} from '@angular/core';
+import {NgControl} from '@angular/forms';
+import {tuiDefaultProp, TuiDestroyService, watch} from '@taiga-ui/cdk';
+import {Observable} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
 
-const SLIDER_INTERACTION_KEYS = new Set([
-    'ArrowLeft',
-    'ArrowRight',
-    'ArrowUp',
-    'ArrowDown',
-    'Home',
-    'End',
-    'PageUp',
-    'PageDown',
-]);
+import {TuiSliderComponent} from './slider.component';
 
 /**
  * Native <input type='range' readonly> doesn't work.
  * This directive imitates this native behaviour.
  */
 @Directive({
     selector: 'input[tuiSlider][readonly]',
+    providers: [TuiDestroyService],
 })
 export class TuiSliderReadonlyDirective {
+    private lastValue = this.slider.value;
+
     @Input()
     @tuiDefaultProp()
     readonly: '' | boolean = true;
 
-    @HostListener('mousedown', ['$event'])
-    @HostListener('touchstart', ['$event'])
-    preventEvent(event: Event) {
-        if (this.readonly === '' || this.readonly) {
-            event.preventDefault();
+    get computedReadonly(): boolean {
+        return this.readonly === '' || this.readonly;
+    }
```

Comment:
what do you think about creating `tuiCoerceBooleanProperty `?

https://material.angular.io/cdk/coercion/api

and then

```
@HostListener('input')
    onInput() {
        if (tuiCoerceBooleanProperty(this.readonly)) {
            this.slider.value = this.lastValue;
        }
    }
```

Reply:
sounds good!)

Reply:
done ✅